### ISSUE TITLE
Detect duplicate notifications

### ISSUE THREAD

**Issue by atymic** [thomasjohnkane/snooze#15]
For example, when a job fails part way through and is retried it's possible to end up with a bunch of duplicate scheduled notifications.

Detecting this and throwing an exception or maybe silent fail?
Should be behind a config option

**Comment by atymic** [comment]
Adding target type/id in #20 will make this much easier.

**Comment by thomasjohnkane** [comment]
So we just check the notification_type and the target_id and the send time right? What is a good buffer time range for predicting duplicates? Same hour?

**Comment by thomasjohnkane** [comment]
Do you think this should happen at creation or send?

**Comment by atymic** [comment]
> What is a good buffer time range for predicting duplicates? Same hour?

I was actually just thinking exact match

> Do you think this should happen at creation or send?

Creation.

**Comment by thomasjohnkane** [comment]
Couldn't this be as simple as change `create` to `firstOrCreate` on line #50 in the ScheduledNotification wrapper? @atymic

**Comment by atymic** [comment]
> Couldn't this be as simple as change create to firstOrCreate on line #50 in the ScheduledNotification wrapper? @atymic

Yep, that would probably work. Just need to make sure were don't apply it to anon notifications. Will PR soon :)

**Comment by thomasjohnkane** [comment]
@atymic - will you remind me why not with anon notifications?

**Comment by atymic** [comment]
In an anon notification the recipient/channel data is inside the serialized target object (which will be different depending on the order, time, etc)
